qoqo_qiskit/src/qoqo_qiskit/backend/backend.py

```diff
@@ -48,40 +54,95 @@ def __init__(
         self.memory = memory
         self.compilation = compilation
 
-    def run_circuit(
+    # Internal _run_circuit method, for async support
+    def _run_circuit(self, circuit: Circuit) -> Tuple[
+        Job,
+        str,
+        Dict[str, int],
+        Dict[str, List[List[bool]]],
+        Dict[str, List[List[float]]],
+        Dict[str, List[List[complex]]],
+    ]:
+        if not isinstance(circuit, Circuit):
+            raise TypeError("The input is not a valid Qoqo Circuit instance.")
+
+        # Initial register setup
+        (
+            clas_regs_sizes,
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        ) = self._setup_registers(circuit)
+
+        # Circuit compilation
+        (compiled_circuit, run_options) = self._compile_circuit(circuit)
+
+        # Error handling
+        self._handle_errors(run_options)
+
+        # Handle simulation Options
+        (shots, sim_type) = self._handle_simulation_options(run_options, compiled_circuit)
+
+        # Job Execution
+        job = self._job_execution(compiled_circuit, shots)
+
+        return (
+            job,
+            sim_type,
+            clas_regs_sizes,
+            output_bit_register_dict,
+            output_float_register_dict,
+            output_complex_register_dict,
+        )
+
+    def _setup_registers(
         self,
         circuit: Circuit,
     ) -> Tuple[
+        Dict[str, int],
         Dict[str, List[List[bool]]],
         Dict[str, List[List[float]]],
         Dict[str, List[List[complex]]],
     ]:
-        """Simulate a Circuit on a Qiskit backend.
-
-        The default number of shots for the simulation is 200.
-        Any kind of Measurement, Statevector or DensityMatrix instruction only works as intended if
-        they are the last instructions in the Circuit.
-        Currently only one simulation is performed, meaning different measurements on different
-        registers are not supported.
+        clas_regs_sizes: Dict[str, int] = {}
 
-        Args:
-            circuit (Circuit): the Circuit to simulate.
+        # Initializing the classical registers for calculation and output
+        internal_bit_register_dict: Dict[str, List[bool]] = {}
+        internal_float_register_dict: Dict[str, List[float]] = {}
+        internal_complex_register_dict: Dict[str, List[complex]] = {}
```

Comment:
These registers seem to be defined and filled in this function, but not passed on. Can they be removed?